packages/dev/core/src/Maths/math.color.ts

```diff
@@ -933,14 +1286,263 @@ export class Color4 {
      * @param result defines the Color4 to fill the result in
      * @returns the result Color4
      */
-    public multiplyToRef(color: Color4, result: Color4): Color4 {
+    public multiplyToRef<T extends this>(color: DeepImmutable<this>, result: T): T {
         result.r = this.r * color.r;
         result.g = this.g * color.g;
         result.b = this.b * color.b;
         result.a = this.a * color.a;
         return result;
     }
 
+    /**
+     * Multiplies in place the current Color4 by the given one.
+     * @param otherColor color to multiple with
+     * @returns the updated Color4.
+     */
+    public multiplyInPlace(otherColor: DeepImmutable<Color4>): this {
+        this.r *= otherColor.r;
+        this.g *= otherColor.g;
+        this.b *= otherColor.b;
+        this.a *= otherColor.a;
+        return this;
+    }
+
+    /**
+     * Returns a new Color4 set with the multiplication result of the given floats and the current Color4 coordinates.
+     * @param r value multiply with
+     * @param g value multiply with
+     * @param b value multiply with
+     * @param a value multiply with
+     * @returns resulting new color
+     */
+    public multiplyByFloats(r: number, g: number, b: number, a: number): this {
+        return new (this.constructor as Constructor<typeof Color4, this>)(this.r * r, this.g * g, this.b * b, this.a * a);
+    }
+
+    /**
+     * Returns a new Color4 set with the division result of the current Color4 by the given one.
+     * @param otherColor color to devide with
+     * @returns resulting new color
+     */
+    public divide(otherColor: DeepImmutable<Color4>): this {
+        return new (this.constructor as Constructor<typeof Color4, this>)(this.r / otherColor.r, this.g / otherColor.g, this.b / otherColor.b, this.a / otherColor.a);
+    }
+    /**
+     * Updates the given color "result" with the division result of the current Color4 by the given one.
+     * @param otherColor color to devide with
+     * @param result color to store the result
+     * @returns result input
+     */
+    public divideToRef<T extends Color4>(otherColor: DeepImmutable<Color4>, result: T): T {
+        result.r = this.r / otherColor.r;
+        result.g = this.g / otherColor.g;
+        result.b = this.b / otherColor.b;
+        result.a = this.a / otherColor.a;
+        return result;
+    }
+
+    /**
+     * Divides the current Color3 coordinates by the given ones.
+     * @param otherColor color to devide with
+     * @returns the updated Color3.
+     */
+    public divideInPlace(otherColor: DeepImmutable<Color4>): this {
+        return this.divideToRef(otherColor, this);
+    }
+
+    /**
+     * Updates the Color4 coordinates with the minimum values between its own and the given color ones
+     * @param other defines the second operand
+     * @returns the current updated Color4
+     */
+    public minimizeInPlace(other: DeepImmutable<Color4>): this {
+        if (other.r < this.r) {

```

Comment:
use min instead if not slower ?

Reply:
Fixed in [157fc18](https://github.com/BabylonJS/Babylon.js/pull/13699/commits/157fc18cd530c9949d920475cb52ee5670b09539)